c2rust/src/bin/c2rust-transpile.rs

```diff
@@ -84,9 +84,18 @@ struct Args {
     #[clap(long = "ddebug-labels")]
     debug_labels: bool,
 
-    /// Input compile_commands.json file
-    #[clap()]
-    compile_commands: PathBuf,
+    /// Input compile_commands.json file or path to source files
+    #[clap(long, parse(from_os_str), required = false, conflicts_with = "source")]
+    compile_commands: Option<PathBuf>,
+
+    /// Source files to process, only valid if compile_commands.json path is not provided.
+    #[clap(
+        long = "source",
+        parse(from_os_str),
+        multiple_values = true,
+        required = false
+    )]
+    source: Vec<PathBuf>,
```

Comment:
```suggestion
    sources: Vec<PathBuf>,
```

If it accepts multiple sources I think the argument name should be plural, too.